get-deps

```diff
@@ -113,47 +139,116 @@ if test -e /etc/arch-release ; then
     'xcb-util-image' \
     'xcb-util-keysyms' \
     'xcb-util-wm'
-  exit $?
+}
+
+bsd_deps() {
+  PKG="$SUDO pkg"
+  $PKG install -y \
+    'cmake' \
+    'curl' \
+    'egl-wayland' \
+    'expat' \
+    'fontconfig' \
+    'gcc' \
+    'gettext' \
+    'git' \
+    'gmake' \
+    'libxcb' \
+    'libxkbcommon' \
+    'mesa-devel' \
+    'openssl' \
+    'p5-ExtUtils-MakeMaker' \
+    'perl5' \
+    'pkgconf' \
+    'python3' \
+    'rust' \
+    'wayland' \
+    'libxcb' \
+    'xcb-util' \
+    'xcb-util-image' \
+    'xcb-util-keysyms' \
+    'xcb-util-wm' \
+    'z' \
+    'zip'
+}
+
+fallback_method() {
+  if test -e /etc/alpine-release; then
+    alpine_deps
+  elif test -e /etc/centos-release || test -e /etc/fedora-release || test -e /etc/redhat-release; then
+    fedora_deps
+  elif test_cmd 'lsb_release' && test "$(lsb_release -si)" = "openSUSE"; then
+    suse_deps
+  elif test -e /etc/debian_version; then
+    debian_deps
+  elif test -e /etc/arch-release; then
+    arch_deps
+  else
+    NOTFOUND=1
+  fi
+}
+
+if test -e /etc/os-release; then
+  . /etc/os-release
 fi
 
+case $ID in
+  centos|fedora|rhel)
+    fedora_deps
+  ;;
+  alpine)
+    alpine_deps
+  ;;
+  *suse*)
+    suse_deps
+  ;;
+  debian|ubuntu)
+    debian_deps
+  ;;
+  freebsd) # available since 13.0
+    bsd_deps
+  ;;
+  arch)
+    arch_deps
+  ;;
+  *)
+    echo "Couldn't find OS by ID, found ID: $ID"
+    echo "Fallback to detecting '/etc/<name>-release'"
+    fallback_method
+    if ! test $NOTFOUND -eq 0; then
+      echo "Couldn't identify OS through '/etc/<name>-release'"
+    fi
+  ;;
+esac
+
 case $OSTYPE in
```

Comment:
SC3028: not defined in POSIX sh

Reply:
Can be ignored, I added check for empty case